grails-app/controllers/au/org/emii/portal/DownloadController.groovy

```diff
@@ -174,10 +174,14 @@ class DownloadController extends RequestProxyingController {
                 if (fieldIndex < currentRow.length) {
 
                     def rowValue = currentRow[fieldIndex].trim()
-                    rowValue = rowValue.replace(prefixToRemove, newUrlBase)
+                    urlSubstitutions.each {
+                        search, replace ->
+
+                        rowValue = rowValue.replaceAll(search, replace)
+                    }
 
                     if (rowValue && includedUrls.add(rowValue)) {
-                        outputWriter.print "$rowValue\n"
```

Comment:
Nah, don't change it unless it turns out to be a problem. And if it is a problem we just fix it and make a comment in the git commit message, or on the line itself) why it is this way. And probably write a test, too.